### ISSUE TITLE
Authentication used on every path even without UseAuthentication() since .NET 8

### ISSUE THREAD

**Issue by fmathgs** [dotnet/aspnetcore#52720]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I have a 
**/api** - Area wich is **protected** and if not covered
**/** - a static **public** website as root.

In .NET 6 the  /api was protected because the corresponding middleware (UseAuthentication() and UseAuthorization()) was added just for this path. This middleware wasn't used for the static website. So the static website was public.

By just switching over to .NET 8 also the static website is using authentication. Even if I never add the middleware in the entire application. Adding the services seems to be enough.

This is totally unexpected and breaks several applications.

### Expected Behavior

Like it was in .NET 6.
The authentication and authorization middleware should only be used if added explicitly. Or at least it should be possible not to use it for specific paths.

### Steps To Reproduce

Run the application from:
https://github.com/fmathgs/auth-middleware-issue-net8

Right now with .NET 8 the root "/" and the api "/api/Main" are using authentication. (which btw. failes as it is not configured correctly for this simple example) => The issue
Change the target framework to .NET 6 and the root "/" is served without any issues and only the api "/api/Main" is using authentication. => As it should be.

### .NET Version

8.0.100

### Anything else?

Somehow related:
https://github.com/dotnet/aspnetcore/issues/48469

**Comment by captainsafia** [comment]
@fmathgs I believe the issue here might be related to the fact that in .NET 7 we added functionality to the minimal hosting that automatically registered authentication and authorization middlewares in the pipeline when their services were registered in the DI container.

I suspect that what is happening here is that you are hitting the automatically registered middlewares before the ones you register in your inner pipeline work.

You can try setting the `__AuthenticationMiddlewareSet` and `AuthorizationMiddlewareSet` properties in your application to disable the default behavior.

```
app.Properties.Add("__AuthenticationMiddlewareSet", true);
app.Properties.Add("__AuthorizationMiddlewareSet", true);
```

Let me know if this ends up working for you.

**Comment by fmathgs** [comment]
Indeed. With those two properties it works now as required and expected. Also in a real application. Thank you.
![image](https://github.com/dotnet/aspnetcore/assets/129957385/5f053544-96c4-487a-9c52-8b29a89c35d3)

Unfortunatelly I didn't find any documentation about this possibility. I'm worried that this "feature" is going to break. Even without a major .NET update. Would be great to get some regular documented option.

**Comment by captainsafia** [comment]
@fmathgs Glad that worked for you! The behavior is currently documented [here](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/middleware?view=aspnetcore-8.0).

We had discussed the possibility of adding a feature flag to disable the behavior when we introduced it but opted not to at the time. We can reconsider this if we get more feedback, especially as more people transition to .NET 8 LTS.
